rogue-thi-app/pages/food.js

```diff
@@ -104,222 +125,323 @@ export default function Mensa () {
-                  id={'allergen-checkbox-' + key}
-                  label={<span><strong>{key}</strong>{' – '}{value}</span>}
-                  checked={allergenSelection[key] || false}
-                  onChange={e => setAllergenSelection({ ...allergenSelection, [key]: e.target.checked })}
-                />
-              ))}
-            </Form>
-          </Modal.Body>
-
-          <Modal.Footer>
-            <Button variant="primary" onClick={() => saveAllergenSelection()}>OK</Button>
-          </Modal.Footer>
-        </Modal>
-      </AppBody>
-
-      <AppTabbar />
-    </AppContainer>
+              </ul>
+
+              <p>
+                <strong>Angaben ohne Gewähr. </strong>
+                <br />
+                Bitte prüfe die Angaben auf den Infobildschirmen, bevor du etwas konsumiert.
+                Die Nährwertangaben beziehen sich auf eine durchschnittliche Portion.
+              </p>
+            </Modal.Body>
+
+            <Modal.Footer>
+              <Button variant="primary" onClick={() => setShowMealDetails(null)}>OK</Button>
+            </Modal.Footer>
+          </Modal>
+
+          <Modal show={showFilterSelection} onHide={() => setShowFilterSelection(false)}>
+            <Modal.Header closeButton>
+              <Modal.Title>Filter</Modal.Title>
+            </Modal.Header>
+
+            <Modal.Body>
+              <Modal.Header>
+                Restaurant
+              </Modal.Header>
+
+              <Form>
+                <Modal.Body>
+                  <Form.Check
+                      id={'restaurant-checkbox-mensa'}
+                      label={<span><strong>Mensa</strong></span>}
+                      checked={selectedRestaurants.includes('mensa')}
+                      onChange={() => toggleSelectedRestaurant('mensa')}
+                  />
+                  <Form.Check
+                      id={'restaurant-checkbox-reimanns'}
+                      label={<span><strong>Reimanns</strong></span>}
+                      checked={selectedRestaurants.includes('reimanns')}
+                      onChange={() => toggleSelectedRestaurant('reimanns')}
+                  />
+                </Modal.Body>
+              </Form>
+
+              <Modal.Header>
+                  <div>
+                    Allergene
+                  </div>
+                  <div>
+                    <Button variant={'outline-primary'} onClick={() => setShowAllergenSelection(true)}>
+                      <FontAwesomeIcon title="Allergene" icon={faAllergies} fixedWidth />
+                    </Button>
+                  </div>
+              </Modal.Header>
+
+              <Modal.Body>
+                <span>
+                  {Object.entries(allergenMap).map(([mapkey, mapvalue]) => (
+                    Object.entries(allergenSelection).map(([key, value]) => (
+                          <span key={key} >{mapkey === key && value === true && mapkey + ' - ' + mapvalue + ', '}</span>
```

Comment:
Here you create <span> elements for all allergenes, even the ones not selected.
I do not think we need those elements at all, a simple comma separated string should be enough.

Something like this:
```javascript
Object.entries(allergenSelection).filter(x => x[1]).map(x => allergenMap[x[0]]).join(', ')
```

Reply:
Will be fixed in next push.